lib/src/test/kotlin/graphql/nadel/engine/NadelDeferSupportTest.kt

```diff
@@ -0,0 +1,208 @@
+package graphql.nadel.engine
+
+import graphql.incremental.DeferPayload
+import graphql.incremental.DelayedIncrementalPartialResult
+import graphql.incremental.DelayedIncrementalPartialResultImpl
+import kotlinx.coroutines.CompletableDeferred
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.channels.Channel.Factory.UNLIMITED
+import kotlinx.coroutines.flow.consumeAsFlow
+import kotlinx.coroutines.flow.lastOrNull
+import kotlinx.coroutines.flow.toList
+import kotlinx.coroutines.test.runTest
+import org.junit.jupiter.api.Test
+import kotlin.test.assertFalse
+import kotlin.test.assertTrue
+
+class NadelDeferSupportTest {
+    @Test
+    fun `channel closes once initial result comes in and there are no pending defer jobs`() {
```

Comment:
Is this description accurate? Wouldn't the channel close only once the defer coroutine scope is complete, which will happen when all pending defer calls are finished?
This description implies that nested defers are not supported, is that the case?

Reply:
So yeah as you discovered later on in the test, nested defer is supported.

This particular description is accurate. Defer calls are meant to be made before the execution result finishes and goes back to the user.

Reply:
i.e. if no defer jobs are pending by the time the first execution job is done, then we can conclude there is no defer. So we close the channel as no other results are coming in.

Also, you effectively cannot submit a defer job _after_ the initial response is done (unless it's nested) because you'd have to be submitting it async to the main coroutine scope of the request.

Reply:
Ah yep, makes sense. Thanks for explaining 👍